libdnf/sack/query.cpp

```diff
@@ -687,6 +730,17 @@ Query::addFilter(int keyname, DnfReldepList *reldeplist)
 int
 Query::addFilter(int keyname, int cmp_type, const char *match)
 {
+    if (keyname == HY_PKG_NEVRA_STRICT) {
+        if (!(cmp_type & HY_EQ))
+            return DNF_ERROR_BAD_QUERY;
+        pImpl->apply();
```

Comment:
Why do you need to apply every time?

Reply:
It is lazy function that is required to initialize query and process all previously added filter. In case that query is already applied, it is automatically skipped.